boards/arm/lora_e5_dev_board/doc/lora_e5_dev_board.rst

```diff
@@ -0,0 +1,278 @@
+| ADC       | on-chip    | adc                                 |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on this Zephyr port.
+
+The default configuration can be found in the defconfig and dts files:
+``boards/arm/lora_e5_dev_board/lora_e5_dev_board_defconfig``
+``boards/arm/lora_e5_dev_board/lora_e5_dev_board.dts``
+
+
+Connections and IOs
+===================
+
+LoRa-E5 Dev Board has 4 GPIO controllers. These controllers are responsible
+for pin muxing, input/output, pull-up, etc.
+
+Available pins:
+---------------
+
+.. image:: img/lora_e5_dev_board_pinout.png
+      :align: center
+      :alt: LoRa-E5 Dev Board Pinout
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+.. rst-class:: rst-columns
+
+- LPUART_1 TX : PC1
+- LPUART_1 RX : PC0
+- USART_1 TX  : PB6
+- USART_1 RX  : PB7
+- USART_2 TX  : PA2
+- USART_2 RX  : PA3
+- I2C_2_SCL   : PB15
+- I2C_2_SDA   : PA15
+- SPI_2_NSS   : PB9
+- SPI_2_SCK   : PB13
+- SPI_2_MISO  : PB14
+- SPI_2_MOSI  : PA10
+- BOOT_PB     : PB13
+- USER_PB     : PA0
+- LED_1       : PB5
+- ADC1 IN2    : PB3
+
+
+Default Zephyr Peripheral to Connector Mapping:
+-----------------------------------------------
+
+.. rst-class:: rst-columns
+
+- RS-485: USART_2
+- grove_serial: USART_1
+- grove_i2c: I2C_2
+
+
+Power Rails
+-----------
+
+The board has multiple power rails, which are always turned on in the default
+configuration.
+
++---------+-------------------+-------------------+
+| Name    | Derived from      | Controlled by     |
++=========+===================+===================+
+| MAIN    | battery, USB, ... | Always on         |
++---------+-------------------+-------------------+
+| VCC     | MAIN              | Always on         |
++---------+-------------------+-------------------+
+| 5V      | MAIN              | SOC pin PB10      |
++---------+-------------------+-------------------+
+| 3V3     | VCC               | SOC pin PA9       |
++---------+-------------------+-------------------+
+
+A list of the devices and their power rails:
+
++--------------------+---------+
+| Device             | Rail    |
++====================+=========+
+| STM32WLE5JC        | VCC     |
++--------------------+---------+
+| RS-485 Transceiver | 3V3     |
++--------------------+---------+
+
+System Clock
+------------
+
+LoRa-E5 Development board System Clock could be driven by the low-power
+internal(MSI), High-speed internal(HSI) or High-speed external(HSE) oscillator,
+as well as main PLL clock.
+By default System clock is driven by the MSI clock at 48MHz.
+
+
+Programming and Debugging
+*************************
+
+Applications for the ``lora_e5_dev_board`` board configuration can be built the
+usual way (see :ref:`build_an_application`).
+
+Be careful to not brick the device by flashing the device before removing the
+flash protections which are initially enabled.
```

Comment:
Hey more of a curiosity than anything else... how can this happens? Thought that the memory write would just fail until the device is erased and unlocked. Is there a condition that bricks it? (apart from locking it permanently)

Reply:
I missed to comment on this one after I addressed the other comments.
To be honest it is not that likely to brick the device as long as you can attach the debugger.
Mainly the option to recover using the system bootloader is no longer directly available.
I have removed this sentence, as the next sentence includes the information about the boot-0 pin not being available.

Reply:
Oh I see... have you actually managed to use a custom firmware with the stock bootloader? Thought you'd had to set up a flash programming offset for that to work. If using the SWD port, the device is shipped with protection enabled, so I don't think you can write to the flash at all without doing a mass erase (that's what I did on mine), maybe worth mentioning that.

Reply:
No, I had no interest in using their bootloader, so I have not tried.
But there is hardly any information available about the bootloader, so I believe it is only supposed to provide them the possibility to upgrade their AT command firmware.

I have rewritten this paragraph to include less information about the bootloader, and instead added a hint that rdp regression must be performed.
Could you please have another look?

Reply:
Yeah looking good, thanks! Is there any step worth documenting to reset the device protection to level 0 with the stlink? I did it manually with a jlink on mine but I presume they have it in their standard tool by now.

Reply:
Thanks. I have added one sentence on how the rdp level can be reset in case an stlink debugger is used.

Reply:
Cool! Saw the update, that's cool, think it'd help a lot for a random user to get started. Only last thing I'd add is a note to call out that the unlock is also going to erase the device, including the preprogrammed bootloader, so the AT firmware cannot be used anymore (unclear to me if Seeed is ever going to distribute the hex files for it).

Reply:
Added a short note that the firmware will be erased:

> The RDP level 1 to RDP level 0 regression will erase the factory programmed AT
> firmware, from which seeed has neither released the source code nor a binary.

Reply:
Cool! Thanks for following up on this!

Reply:
Thanks for the recommendations.